airbyte-integrations/connectors/source-cart/source_cart/spec.json

```diff
@@ -12,16 +12,23 @@
         "airbyte_secret": true,
         "description": "API Key. See the <a href=\"https://docs.airbyte.io/integrations/sources/mailchimp\">docs</a> for information on how to generate this key."
       },
+      "store_name": {
+        "type": "string",
+        "description": "Store name. All API URLs start with https://[mystorename.com]/api/v1/, where [mystorename.com] is the domain name of your store."
+      },
       "start_date": {
         "title": "Start Date",
         "type": "string",
         "description": "The date from which you'd like to replicate the data",
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         "examples": "2021-01-01T00:00:00Z"
       },
-      "store_name": {
+      "end_date": {
```

Comment:
There is also an alternative implementation:

1. In paypal source we agreed to not define it in spec file.
We just set "additionalProperties": True, in spec file 
and just added end_date param in tests related config files. (so it is not visible for users)

2. Will 'current' users be able to use theirs 'old' spec files without end date attr there?
3. Mostly, users will not use this option

4. So probably, we should discuss current and alternative approach with Sherif

Reply:
I agree with @midavadim, we should have the hidden  parameter inside  the test config file, but leave the spec as it was before, the downside is the presence of the end_date inside of the code, it could be confusing for future support, but leaving the proper comments - is the way it should be.

Reply:
We can also go with the next approach: 
- leave the `end_date` optional, if the customer doesn't supply`end date` inside `spec` that's fine - we would use the `now()` instead, otherwise - `end_date` from the `spec`. Therefore we leave a comment for the customer `"By leaving the end_date empty, the replication period will be till present time"`

WDYT?

Reply:
@midavadim , I tried to use this option without the spec changes (with additionalProperties=True only) but base tests were broken.
Sure we should ask Airbyte team about this case

Reply:
I do remember we tried to use 'additional' properties in specm but it just required additionalProperties=True

Reply:
Once 'Will 'current' users be able to use theirs 'old' spec files without end date attr there?' - is not a problem, then I 100% fine with this approach.

rephrase - will the customer be able to run syncs with old specs on this updated source?

Reply:
@midavadim ,
Yes, this attr is optional and doesn't have any default value. and this condition will be validated this:
```
if self._end_date and self._end_date > start_date:
  pass
```

Reply:
since we are introducing this only to make testing easier I would prefer to make it a hidden option like suggested above. Is there any blocker to doing this? (asking because you mentioned needing to talk w airbyet about something)